### DISCUSSION TITLE
[GITHUB ACTIONS] - actions/upload-artifact

### DISCUSSION THREAD

**Discussion by RachelXGanon** [community/community#28030] [Actions] [answered]
Hi, 
I'm using [github action upload artifact](https://github.com/actions/upload-artifact), 
the action generates a ZIP file that can be downloaded by clicking the name.
I'd like to sign with digital signature the ZIP, by the pipeline.
How can I find the path of the ZIP file?(where's it store?)
Than  you.

**Comment by airtower-luna** [comment, marked as answer]
As I understand the [Zipped Artifact Downloads section](https://github.com/actions/upload-artifact#zipped-artifact-downloads) of the `actions/upload-artifact` documentation the ZIP isn't created until the workflow is done:

> During a workflow run, files are uploaded and downloaded individually using the upload-artifact and download-artifact actions. However, when a workflow run finishes and an artifact is downloaded from either the UI or through the [download api](https://developer.github.com/v3/actions/artifacts/#download-an-artifact), a zip is dynamically created with all the file contents that were uploaded.

In that case it's probably best if you sign the files individually, or (easier to verify) create a checksum file and sign that.

I hope that helps!

**Reply by RachelXGanon** [reply]
It definitely did. 
Thank you

**Reply by airtower-luna** [reply]
I'm glad to hear! In that case marking the answer would be appreciated. :smiley_cat:

**Reply by RachelXGanon** [reply]
Sure, done :)
